tests/test_validate.py

```diff
@@ -768,3 +768,94 @@ def test_that_standardize_string_for_validation_returns_expected(
         allow_underscore=allow_underscore,
     )
     assert test_str == expected
+
+
+@pytest.mark.parametrize(
+    "input,expected_index,allowed_alleles,ignore_case",
+    [
+        (
+            pd.DataFrame(
+                {"REFERENCE_ALLELE": ["ACGT-G", "A-CGT ", "A", "C", "T", "G", "-", " "]}
```

Comment:
```suggestion
                {"REFERENCE_ALLELE": ["ACGTG", "ACGT ", "A", "C", "T", "G", "-", " "]}
```

lets make the white space invalid.  Lets try and rely on pandas as much as possible, if pandas reads in the " "  as float('nan') then it's fine to include it here.

Reply:
Yes, thankfully `pandas.str.match` has a parameter that handles anything NA so we should be good on that front and don't need to add the long list of possible NA values here.